### ISSUE TITLE
lazyhash field hashcode needs @SuppressWarnings("immutable")

### ISSUE THREAD

**Issue by schallee** [immutables/immutables#1114]
When using `@Value.Immutable(lazyhash=true) ` the generated field `private transient int hashCode` is causing compilation with  errorprone to warn:

> [Immutable] type annotated with @Immutable could not be proven immutable: '...' has non-final field 'hashCode'

This needs to have `@SuppressWarnings("immutable")` added. I think this just needs ` [hiddenMutableState type]` added to`Immutables.generator`.

Unfortunately I'm not familiar with the template system so can't provide a PR.

**Comment by asereda-gs** [comment]
pls let me know if b110e21 works for you

**Comment by schallee** [comment]
Sorry for the slow reply on my end. Yes it does indeed work. Thank you and others for the great package!

It took me until this morning to get it to compile so I could try it. On a hunch I tried using an Oracle JDK instead of an IBM JDK and that built the first time (both 1.8 on debian linux amd64). Is there any reason you're aware of issues between the two? I'm thinking to put in a separate issue for this so others don't spend the frustration I did as I've seen others having issues with compilation as well. I'll hold off though for the moment as perhaps this is a known issue and I just missed it.

Again, thanks for all your work on immutables!
